agent/taskresource/grpcclient/credentialsfetcherclient_test.go

```diff
@@ -111,6 +131,105 @@ func TestCredentialsFetcherClient_AddKerberosLease(t *testing.T) {
 	}
 }
 
+func TestCredentialsFetcherClient_AddNonDomainJoinedKerberosLease(t *testing.T) {
+	tests := []struct {
+		name             string
+		credspecContents []string
+		username         string
+		password         string
+		domain           string
+		response         CredentialsFetcherResponse
+	}{
+		{
+			"invalid request empty credspec contents",
+			[]string{},
+			"testusername",
+			"testpassword",
+			"testdomain",
+			CredentialsFetcherResponse{},
+		},
+		{
+			"invalid request username, password or domain should not be empty",
+			[]string{credspec_webapp01},
+			"",
+			"",
+			"",
+			CredentialsFetcherResponse{},
+		},
+		{
+			"valid request credspecs associated to gMSA account",
+			[]string{credspec_webapp01},
+			"testusername",
+			"testpassword",
+			"testdomain",
+			CredentialsFetcherResponse{LeaseID: leaseid, KerberosTicketPaths: []string{"/var/credentials-fetcher/krbdir/123456/webapp01", "/var/credentials-fetcher/krbdir/123456/webapp02"}},
+		},
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.DialContext(ctx, "", grpc.WithInsecure(), grpc.WithContextDialer(dialer()))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer conn.Close()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			response, _ := NewCredentialsFetcherClient(conn, time.Minute).AddNonDomainJoinedKerberosLease(context.Background(), tt.credspecContents, tt.username, tt.password, tt.domain)
```

Comment:
Can we please split this long line into shorter lines for better readability. 

```go
response, _ := NewCredentialsFetcherClient(conn, time.Minute)
    .AddNonDomainJoinedKerberosLease(
        context.Background(), tt.credspecContents, tt.username, tt.password, tt.domain,
    )
```

Reply:
Also let's not ignore the error returned by the `AddNonDomainJoinedKerberosLease` call. We are using an older `err` in the assertions below.